src/OpenLoco/Vehicles/VehicleHead.cpp

```diff
@@ -824,10 +824,158 @@ namespace OpenLoco::Vehicles
+                if (var_36 != train.veh2->var_36 || train.veh2->var_2E != var_2E)
+                {
+                    tryCreateInitialMovementSound();
+                    return true;
+                }
+                return sub_4A8F22();
+            }
+            else
+            {
+                tryCreateInitialMovementSound();
+                return true;
+            }
+        }
+        else
+        {
+            if (al == 4)
+            {
+                status = Status::approaching;
+                stationId = nextStation;
+                tryCreateInitialMovementSound();
+                return true;
+            }
+            else if (al == 3)
+            {
+                if (train.veh2->var_36 != var_36 || train.veh2->var_2E != var_2E)
+                {
+                    tryCreateInitialMovementSound();
+                    return true;
+                }
+
+                status = Status::unk_3;
+
+                auto* vehType1 = train.veh1;
+                vehType1->var_46++;
+
+                if (var_0C & Flags0C::manualControl)
+                {
+                    var_5C = 2;
+                    vehType1->var_48 |= 1 << 0;
+                    tryCreateInitialMovementSound();
+                    return true;
+                }
+
+                if (flags & (1 << 1))
+                {
+                    if (vehType1->var_46 < 1920)
+                    {
+                        tryCreateInitialMovementSound();
+                        return true;
+                    }
+
+                    if (!(flags & (1 << 7)))
+                    {
+                        if (sub_4AC1C2())
+                        {
+                            var_5C = 2;
+                            vehType1->var_48 |= 1 << 0;
+                            tryCreateInitialMovementSound();
+                            return true;
+                        }
+                    }
+                    return landReverseFromSignal();
+                }
+                else
+                {
+                    if (!(vehType1->var_46 & 0x3F))
+                    {
+                        if (!(flags & (1 << 7)))
+                        {
+                            if (sub_4AC1C2())
+                            {
+                                var_5C = 2;
+                                vehType1->var_48 |= 1 << 0;
+                                tryCreateInitialMovementSound();
+                                return true;
+                            }
+                        }
+
+                        if (sub_4AC0A3())
+                        {
+                            return landReverseFromSignal();
+                        }
+                    }
+
+                    if (vehType1->var_46 < 640)
+                    {
+                        tryCreateInitialMovementSound();
+                        return true;
+                    }
+                    return landReverseFromSignal();
```

Comment:
I think this would be more readable with the logic reversed. Perhaps with a constant for the 640 as well.